### PR TITLE
[TRTLLM-14839][chore] Relocate root PEFT modules into _torch/peft/

### PR DESCRIPTION
**Pull request by lori-ren**
## Dev Engineer Review

- Relocated LoRA configuration, checkpoint loaders, runtime manager, and prompt-adapter modules into `_torch/peft/`.
- Updated production imports, documentation, examples, serialization allowlists, CODEOWNERS, and formatting configuration.
- Preserved runtime behavior and API compatibility through import-path updates.
- Added public loader utilities and `HfLoraLoader` and `NemoLoraLoader`.
- Removed loader implementations from `manager.py` after moving them to `loaders.py`.
- Updated runtime comments and docstrings to reference the new paths.
- Added extensive MoE LoRA and FP8 LoRA regression coverage.
- Configuration and test-list paths remain consistent with the new `_torch/peft/` layout.
- Shim modules are not included. A follow-up is required to preserve legacy import paths.

## QA Engineer Review

- Added or relocated tests under `tests/unittest/_torch/peft/`.
- Added FP8 LoRA regression tests, including:
  - `test_fp8_eager_grouped_gemm_matches_reference`
  - `test_fp8_cuda_graph_grouped_gemm_matches_reference_after_replay`
  - FP8 cache, dtype, alignment, dispatch, metadata, and split-K tests.
- Added LoRA state-management tests in `test_lora.py`.
- Added MoE layout tests in `test_moe_layout.py`.
- Added MoE CUDA-graph parameter tests in `test_moe_lora_cuda_graph_params.py`.
- Added MoE extraction, grouped-GEMM, model-path, operator, and validator tests.
- Updated existing tests to import PEFT modules from their new locations.
- Updated CI test lists:
  - `tests/integration/test_lists/test-db/l0_cpu.yml`
  - `tests/integration/test_lists/test-db/l0_h100.yml`
- The updated test lists select `unittest/_torch/peft` instead of `unittest/_torch/lora`.
- Verdict: sufficient.

## Description

Move LoRA and prompt-adapter loading out of the package root into the existing Torch PEFT package:
- lora_helper.py -> _torch/peft/lora/config.py
- lora_manager.py -> _torch/peft/lora/loaders.py for checkpoint loading +  _torch/peft/lora/manager.py for runtime adapter management
- prompt_adapter_manager.py -> _torch/peft/prompt_adapter.py

Tests are relocated as well to match the implementation directory name.

Follow-up: shim files are *intended* to be absent in this PR, so as to prevent git blaming moved files to PR author. The shim files shall be created in a follow-up PR.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -121,11 +121,8 @@
 /tensorrt_llm/inputs @NVIDIA/trt-llm-runtime-devs
 /tensorrt_llm/llmapi @NVIDIA/trt-llm-runtime-devs
 /tensorrt_llm/llmapi/mm_encoder.py @NVIDIA/trt-llm-runtime-devs @NVIDIA/trt-llm-models-devs
-/tensorrt_llm/lora_helper.py @NVIDIA/trt-llm-runtime-devs
-/tensorrt_llm/lora_manager.py @NVIDIA/trt-llm-runtime-devs
 /tensorrt_llm/mapping.py @NVIDIA/trt-llm-runtime-devs
 /tensorrt_llm/metrics @NVIDIA/trt-llm-runtime-devs
-/tensorrt_llm/prompt_adapter_manager.py @NVIDIA/trt-llm-runtime-devs
```
Comment (lori-ren) [inline .github/CODEOWNERS:128]: To reviewers: These lines are indented to be removed. Now all these files are placed under `_torch/peft`, which itself has already been included in owner list. We would add these files back in a separated shim section in the follow-up shim PR.

File: tensorrt_llm/_torch/models/modeling_utils.py [hunk 2]
```diff
@@ -17,8 +17,8 @@
 from torch.utils._pytree import tree_any_only
 from tqdm import tqdm
 
+from tensorrt_llm._torch.peft.lora.loaders import HfLoraLoader
```
[the remark this answers is not in the document]
Comment (lori-ren) [reply tensorrt_llm/_torch/models/modeling_utils.py:20]: As I wrote in PR description, the shim files are intended to be absent, since if we add shim files in this PR, git blame will track all related files to me, instead of original code owner. Therefore, we have got to create two PRs (one for migration and one for shim files) since main branch only allows squash-merge (which condense all commits into one). The migration itself stays api-compatible.

**Review by QiJune — APPROVED** [review]
LGTM

**Review by chzblych — APPROVED** [review]

**Review by Shixiaowei02 — APPROVED** [review]

File: tensorrt_llm/llmapi/llm_args.py [hunk 3]
```diff
@@ -43,9 +43,9 @@
 except ImportError:
     PlacementGroup = None
 
+from tensorrt_llm._torch.peft.lora.config import (
```
Comment (BowenFu) [inline tensorrt_llm/llmapi/llm_args.py:46]: These legacy import paths are public and are also used by serialized payloads. Please include the compatibility shims in this PR so `tensorrt_llm.lora_helper`, `tensorrt_llm.lora_manager`, and `tensorrt_llm.prompt_adapter_manager` remain usable at every merged head; a later follow-up leaves an externally visible break between merges.

Comment (lori-ren) [reply tensorrt_llm/llmapi/llm_args.py:46]: We are not able to do that in a single PR, since main branch only accepts squash-merge policy, which will condense all commits into one and results in git blaming all related files to author. The only possible way is to keep two PRs and merge them as soon as possible.

Do you have any suggestions?

**Review by fredricz-20070104 — COMMENTED** [review]
## Review summary - CONCERNS
**Verdict:** This is a clean mechanical relocation and can merge once one import-boundary risk is confirmed; correctness of the move hinges on manager.py not referencing symbols it no longer imports.

### Concerns
1. **[MAJOR] `tensorrt_llm/_torch/peft/lora/manager.py:40` - possible reference to moved symbols that are no longer imported**
   - *What is wrong*: The split moved `iterate_hf_lora`, `get_hf_target_modules`, `norm_dora_magnitude` and `HF_LORA_PATTERN` out to `loaders.py`, but manager.py's new `from .loaders import (...)` block re-imports only `HfLoraLoader`, `NemoLoraLoader`, `find_nemo_files`, `get_all_hf_lora_weights`, `get_all_nemo_lora_weights`, `invert_module_mapping`, `unpack_nemo_weights`, `NEMO_SUPPORTED_LORA_MODULES`. The `LoraManager` class body and `load_torch_*` functions are not in the diff, so I cannot see whether they still use one of the un-imported names.
   - *How it fails*: In the original file `norm_dora_magnitude` is used on the DoRA weight-normalization path inside LoraManager. If that (or `HF_LORA_PATTERN`/`iterate_hf_lora`) is still referenced in manager.py, loading a DoRA/HF adapter raises `NameError` at runtime -- import-time smoke tests will not catch it, only a test that actually loads such an adapter.
   - *Suggested fix*: `grep -n "norm_dora_magnitude\|iterate_hf_lora\|get_hf_target_modules\|HF_LORA_PATTERN" tensorrt_llm/_torch/peft/lora/manager.py`; add any hits to the `.loaders` import, and make sure a DoRA/NeMo load test runs.

### Minor notes (non-blocking)
- `tensorrt_llm/llmapi/serialization.py:110` - allowlist now points LoraConfig at the new module path; objects pickled by an older build (`tensorrt_llm.lora_manager.LoraConfig`) will no longer deserialize since no shim preserves the old path.

### QA view
- **Test coverage:** partial - test files are relocated alongside the code and `tests/unittest/others/test_lora_manager.py` exercises the new `loaders`/`manager` boundary for `HfLoraLoader`+`LoraManager`, but nothing in the diff shows the DoRA path (the one at risk above) being run.
- **SM coverage:** architecture-independent - no `get_sm_version`, arch guards, or fp8/nvfp4 code changed; the FP8 LoRA test file is a rename with no content change.
- **Test code:** no issues introduced - test changes are pure renames or single-line import swaps.
- **Test time:** none - renames and import swaps only; the test-list edits swap `unittest/_torch/lora` for `unittest/_torch/peft` (same test set, new dir).
- **Needs `/qa-verify`:** yes - large relocation with a new import boundary plus deliberate removal of legacy shims; QA should confirm no NameError on the DoRA/HF/NeMo load paths, no lingering callers of the removed top-level modules, and that LoraConfig still deserializes.

### Possible new issues
- **No shims / API break**: `tensorrt_llm.lora_manager`, `tensorrt_llm.lora_helper`, and `tensorrt_llm.prompt_adapter_manager` are deleted with no shim. Any external or internal caller not among the 52 changed files breaks on import. This is documented as intentional (shim follow-up PR), but the PR is not labelled BREAKING - consider adding the label.
- **Import cycle risk**: manager.py now lives under `_torch/peft/lora` yet imports `tensorrt_llm.bindings/mapping/models.convert_utils` and is imported by `resource_manager`/`model_engine`/`base_worker`; a new cycle would fail broadly (CI would catch it), but it is a relocation-specific hazard worth a sanity import.

### What I could not verify
- The bodies of `LoraManager` and `load_torch_*` in manager.py are not in the diff, so the MAJOR finding is a verification request, not a confirmed defect.
- Whether every repo-wide caller of the three removed modules is covered by the 52 changed files (grep of the full tree needed).

_Automated review by NVCortex Lite, run by @fredricz-20070104._

**Review by fredricz-20070104 — APPROVED** [review]
## Review summary - Approve (non-blocking)

Approving so this is not blocked on me. The points raised in my review comment above are **non-blocking** — please read them and address what you agree with before merging.

Worth doing before this is relied on: Large module relocation with a new import boundary (manager.py <- loaders.py) plus deliberate removal of legacy import shims. QA should confirm: (1) manager.py has no NameError on the DoRA / HF / NeMo load paths, (2) no remaining internal callers of the removed top-level modules, and (3) LoraConfig still deserializes via the updated allowlist.

_Automated review by NVCortex Lite, run by @fredricz-20070104._

**Review by sunnyqgg — APPROVED** [review]
LGTM for MoE part

File: tensorrt_llm/llmapi/llm_args.py (hunk 3 above)
Comment (BowenFu) [reply tensorrt_llm/llmapi/llm_args.py:46]: Could we keep `tensorrt_llm/lora_helper.py`, `tensorrt_llm/lora_manager.py`, and `tensorrt_llm/prompt_adapter_manager.py` in this PR as small re-export shims that pull from the new location and emit a `DeprecationWarning`, and drop them in the follow-up once downstream imports have moved?

On the blame concern: squash-merge collapses the relocation into one commit either way, so splitting across two PRs doesn't preserve blame on the moved files — the history is condensed regardless. The shims would be new files, so they wouldn't touch the blame of anything you relocated, and every merged head would stay importable for external users and for unpickling serialized payloads.

Required for this PR, I think — the concern is the externally visible break at the intermediate merged head rather than the file layout itself.

Comment (lori-ren) [reply tensorrt_llm/llmapi/llm_args.py:46]: No, if we split the PR into two commits, git won't blame the moved file to PR author, while a squashed PR does. This was tested before we decided to split the refactor into two PRs.

The shim files are intended to be removed, however, removal of these files would not recover the original author info.

This is disturbing, but we have no choice.

Comment (BowenFu) [reply tensorrt_llm/llmapi/llm_args.py:46]: You're right, and I'm withdrawing the request — I tested it rather than take it on trust. On a pure rename git detects the rename and `git blame` on the new path keeps the original authors. If the same commit also leaves a shim at the old path, the rename is no longer detected (git records it as an add plus a modify) and blame on the new file attributes every line to the relocating commit. So folding the shims into this PR would cost exactly what you described, and permanently. Sorry for pushing on it twice.

The one thing I'd still like is the window bounded rather than open-ended. #17857 is currently a draft with no reviews, so "as soon as possible" isn't yet backed by a mergeable PR. Could we link it from this description and take it out of draft, so the two can go in back to back?

Non-blocking — I'm not holding this PR on the sequencing.

**Review by BowenFu — APPROVED** [review]
**Approving** — my public-interface concern is withdrawn, and I checked the relocation itself rather than relying on the other approvals.

- git detects all 12 renames, so `blame` attribution survives on `config.py`, `manager.py`, `prompt_adapter.py` and the relocated tests
- `loaders.py` is a genuine content move out of `lora_manager.py` — the only lines that are not carried over verbatim are the license header, the module docstring, and imports rewritten from relative to absolute
- nothing dropped in the split; `_FP8_LORA_TMA_ALIGNMENT` is repositioned within `manager.py`, not lost

The CodeRabbit findings in `loaders.py` (`self.lora_weight` init, the bare `Exception`) are pre-existing code being moved, so they are not this PR's to close.

Still non-blocking, as noted in the thread: linking #17857 and taking it out of draft would bound the window between the two merges.

**Review by BowenFu — APPROVED** [review]

File: tensorrt_llm/llmapi/llm_args.py (hunk 3 above)
Comment (lori-ren) [reply tensorrt_llm/llmapi/llm_args.py:46]: The question is that, the shim PR must be created on the basis of current PR, which is still under review. If we change the state of draft shim PR to ready for review, it will ping exactly same code owners of current PR, since it contains all changes of current PR. Consequently, we need to perform a rebase to shim PR as soon as current PR gets merged before we change its state to ready, or we have to ask code owners for a second review, which may cost even longer time...
